### PR TITLE
Added tests for Book and Category layers.

### PR DESCRIPTION
**Pull request by k0sm0naft**
image

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -112,8 +101,41 @@
...
+			<artifactId>spring-boot-testcontainers</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.testcontainers</groupId>
+			<artifactId>mysql</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
```
Comment (vovasalyha) [inline pom.xml:120]: Mockito is a part of spring-boot-starter-test, you can remove it.

File: src/test/java/ua/bookstore/online/controller/BookControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        tearDown(dataSource);
+    }
+
+    @Test
+    @DisplayName("Create new book")
+    @WithMockUser(username = "admin", roles = {"MANAGER"})
+    void createBook_CreateNewBook_ReturnsExpectedBook() throws Exception {
```
Comment (vovasalyha) [inline src/test/java/ua/bookstore/online/controller/BookControllerIntegrationTest.java:87]: The name of the test method does not correspond to what is tested in the method (I see that this test checks that conflict response is received if ISBN is not unique)

File: src/test/java/ua/bookstore/online/service/impl/BookServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+        // When
+        BookDto actual = bookService.save(requestDto);
+
+        // Then
+        assertNotNull(actual);
+        BookDto expected = getBookDto(book);
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+
+        // Verify repository method calls
+        verify(bookRepository).findByIsbn(requestDto.isbn());
```
Comment (vovasalyha) [inline src/test/java/ua/bookstore/online/service/impl/BookServiceImplTest.java:83]: It's not necessary to verify execution of methods which you previously mocked by `when().then()`. `verify` is only useful when you don't mock the method to return desired result but simply want to verify that method was actually called. For more details you can check the javadoc of `verify` method.

**Review by vovasalyha — APPROVED** [review]
GJ. Left few comments to consider, but nothing critical.

File: src/test/java/ua/bookstore/online/controller/BookControllerIntegrationTest.java (hunk 2 above)
Comment (k0sm0naft) [reply src/test/java/ua/bookstore/online/controller/BookControllerIntegrationTest.java:87]: I will do it in the next PR.

**Pull request merged**